gradle-baseline-java/src/main/java/com/palantir/baseline/plugins/BaselineModuleJvmArgs.java

```diff
@@ -158,111 +156,148 @@ public void execute(Manifest manifest) {
-                                                    extension.exports().get().stream(),
-                                                    extension.opens().get().stream()),
-                                            info.stream()
-                                                    .flatMap(item -> Stream.concat(
-                                                            item.exports().stream(), item.opens().stream())))
-                                    .distinct()
-                                    .sorted()
-                                    .map(item -> item + "=ALL-UNNAMED")
-                                    .collect(ImmutableList.toImmutableList());
-                            log.debug(
-                                    "BaselineModuleJvmArgs building {} on {} with exports: {}",
-                                    javadoc.getName(),
-                                    project.getPath(),
-                                    exportValues);
-                            if (!exportValues.isEmpty()) {
-                                coreOptions
-                                        // options are automatically prefixed with '-' internally
-                                        .addMultilineStringsOption("-add-exports")
-                                        .setValue(exportValues);
-                            }
-                        } else {
-                            log.error(
-                                    "MinimalJavadocOptions implementation was " + "not CoreJavadocOptions, rather '{}'",
-                                    options.getClass().getName());
-                        }
+        project.getTasks().named(sourceSet.getJavadocTaskName(), Javadoc.class, javadoc -> {
+            // Javadoc task is horrible. There's no lazy properties/CommandLineArgumentProviders, so we have to
+            // resort to this fresh hell of changing its configuration in a task action before it runs. Once
+            // Gradle have made this lazy, this should be rewritten.
+            javadoc.doFirst(new Action<Task>() {
+                @Override
+                public void execute(Task _task) {
+                    MinimalJavadocOptions options = javadoc.getOptions();
+                    if (!(options instanceof CoreJavadocOptions coreOptions)) {
+                        log.error(
+                                "MinimalJavadocOptions implementation on '{}' was not CoreJavadocOptions, rather '{}'",
+                                javadoc.getPath(),
+                                options.getClass().getName());
+                        return;
                     }
-                });
 
-                setTaskInputsFromExtension(javadocTask, extension);
+                    Set<String> exportValuesRaw = ModuleJvmArgsArgumentProvider.fromJustExtensionForCompilation(javadoc)
+                            .configureWithClasspath(sourceSet::getAnnotationProcessorPath)
```

Comment:
talking IRL, don't necessarily need to do this? but requires a test as its a change in behaviour